XMonad/Layout/LimitWindows.hs

```diff
@@ -1,4 +1,7 @@
 {-# LANGUAGE CPP, FlexibleInstances, MultiParamTypeClasses, DeriveDataTypeable, PatternGuards #-}
+#ifdef TESTING
+{-# OPTIONS_GHC -Wno-duplicate-exports #-}
+#endif
```

Comment:
Is there any reason we shouldn't just expose those internals (maybe under a literal `*internals` heading)?  This way we'd get rid of the CPP

Reply:
Many people (including me) import xmonad-contrib modules unqualified so exposing stuff named `select` and `update` might lead to extra duplication warnings.

If we really wanted to get rid of CPP it would probably be better to move that stuff to a separate X.L.LimitWindows.Internal module.

Reply:
Ah good point.  We have no precedence of an `Internal` style module while `CPP` is (sadly, at least imo) used all over the place so it's probably fine as it is now

Reply:
I am not opposed to going with Internal later (possibly as part of a greater effort to silence warnings), but right now I wanted a quick fix to make it bearable to iterate on the tests in ghcid.